plugins/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,579 @@
+                    // the response indicated the "Connection: close" semantic.
+                    return derived().do_eof();
+                }
+
+                // create a new parser to clear state 
+                req_parser_.emplace();
+                req_parser_->body_limit(plugin_state_->max_body_size);
+                // create a new response object
+                res_.emplace();
+
+                // Read another request
+                do_read();
+            }           
+
+            virtual void handle_exception() override {
+                std::string err_str;
+                try { 
+                    throw;
+                } catch(const fc::exception& e) {
+                    err_str = e.to_detail_string();
+                    fc_elog( logger, "fc::exception: ${w}", ("w", err_str) );
+                } catch(std::exception& e) {
+                    err_str = e.what(); 
+                    fc_elog( logger, "std::exception: ${w}", ("w", err_str) );
+                } catch( ... ) {
+                    err_str = "unknown";
+                    fc_elog( logger, "unkonwn exception");
+                } 
+
+                if(is_send_exception_response_) {
+                    res_->set(http::field::content_type, "text/plain");
+                    res_->keep_alive(false);
+                    res_->set(http::field::server, BOOST_BEAST_VERSION_STRING);
+
+                    http::status stat = http::status::internal_server_error;
+                    auto resp_str = "Internal Server Error\n\nUnhandled Exception: " + err_str;
+                    send_response(resp_str, static_cast<int>(stat));
+                    derived().do_eof();
+                }
+            }
+
+            virtual void send_response(std::optional<std::string> body, int code) override {
+                write_begin_ = steady_clock::now();
+                auto dt = write_begin_ - handle_begin_;
+                handle_time_us_ += std::chrono::duration_cast<std::chrono::microseconds>(dt).count();
+                
+                res_->result(code);
+                if(body.has_value())
+                    res_->body() = *body;        
+
+                res_->prepare_payload();
+
+                // Determine if we should close the connection after
+                bool close = !(plugin_state_->keep_alive) || res_->need_eof();
+
+                // Write the response
+                auto self = derived().shared_from_this();
+                http::async_write(
+                    derived().stream(),
+                    *res_,
+                    [self, close](beast::error_code ec, std::size_t bytes_transferred) {
+                        self->on_write(ec, bytes_transferred, close);
+                    }
+                );
+            }
+
+            void run_session() {
+                // wrap entiere session around try/catch to prevent non-zero exit code test failure
```

Comment:
This comment does not seem relevant. Should we add a try-catch here to be safe?

Reply:
It should not be necessary.   There isn't much code between that point and either an  async_read will happen, which cannot throw an exception, or session will be terminated, if max_requests_in_flight is exceeded.